src/assets/scripts/client/trafficGenerator/SpawnPatternModel.js

```diff
@@ -312,26 +477,106 @@ export default class SpawnPatternModel extends BaseModel {
     }
 
     /**
-     * Returns a random whole number between the allowed min and max delay.
+     * Return a number to use for the next delay period calculated based
+     * on spawnning method.
      *
      * This is the value that will be used by the `SpawnScheduler` when
      * when creating a new spawn interval.
      *
      * @for SpawnPatternModel
-     * @method getRandomDelayValue
-     * @return randomDelayValue {number}
+     * @method getNextDelayValue
+     * @param gameTime {number}
+     * @return {number}             Next delay period based on spawn method
      */
-    getRandomDelayValue() {
-        // TODO: make this method a fascade by implementing a switch to handle #method variations with other class methods
-        let targetDelayPeriod = this._maximumDelay;
+    getNextDelayValue(gameTime = 0) {
+        switch (this.method) {
+            case SPAWN_METHOD.RANDOM:
+                return this._calculateRandomDelayPeriod();
+            case SPAWN_METHOD.CYCLIC:
+                return this._calculateNextCyclicDelayPeriod(gameTime);
+            case SPAWN_METHOD.SURGE:
+                return this._calculateNextSurgeDelayPeriod(gameTime);
+            case SPAWN_METHOD.WAVE:
+                return this._calculateNextWaveDelayPeriod(gameTime);
```

Comment:
Don't you want `break` statements within each case, so it doesn't run all of the cases?